subprojects/model-core/src/main/java/org/gradle/internal/reflect/annotations/impl/DefaultTypeAnnotationMetadataStore.java

```diff
@@ -83,6 +88,12 @@ public void visitValidationFailures(@Nullable String ownerPath, ParameterValidat
         }
     };
 
+    private static final ImmutableSet<? extends Type> MUTABLE_NON_FINAL_TYPES = ImmutableSet.of(
+        ConfigurableFileCollection.class,
+        DomainObjectCollection.class,
```

Comment:
We should remove`DomainObjectCollection` for now I think. If we leave it in we should also include it in the tests.

Reply:
Indeed, done